### ISSUE TITLE
Are filenames with hash/number sign # supported?

### ISSUE THREAD

**Issue by stax76** [gpac/gpac#831]
Are filenames with hash/number sign # supported?

**Comment by aureliendavid** [comment]
Hm, 

It should be ok for output files. However for input files the # sign is used to extract tracks. Some tests:

```
$ MP4Box -add file.mp4 -new "out#hash.mp4"
IsoMedia import file.mp4 - track ID 1 - Video (size 1280 x 720)
IsoMedia import file.mp4 - track ID 2 - Audio (SR 44100 - 1 channels)
Saving out#hash.mp4: 0.500 secs Interleaving
```

ok for output file

```
$ MP4Box -add "file#hash.mp4" -new test.mp4
[Importer] Unknown input file type for "file"
Error importing file#hash.mp4: Requested URL is not valid or cannot be found
```

error on an input file as it tries to open `file` with the option `hash.mp4`

```
$ MP4Box -add "file#hash.mp4#video" -new test.mp4
IsoMedia import file#hash.mp4 - track ID 1 - Video (size 1280 x 720)
Saving test.mp4: 0.500 secs Interleaving
```

however this one works because the first hash sign is ignored since another one is present


I guess a possible fix would be to look for the hash sign that extracts tracks only after the file extension. It would fix the majority of use cases.

**Comment by stax76** [comment]
> I guess a possible fix would be to look for the hash sign that extracts tracks only after the file extension. It would fix the majority of use cases.

what if the filename is like a.#.b.mp4

**Comment by aureliendavid** [comment]
it should be ok as long as we detect the extension properly (the remaining problem would be files without an extension but a # in the name but that should be very rare)

i'll try to push a fix soon

**Comment by jeanlf** [comment]
fixed in 0.9.0 dev
